### PR TITLE
Added 5yr, scheduled resolution date, and bot filters to track record. Related refactors

### PR DESCRIPTION
**Pull request by CodexVeritas**
Issue: https://github.com/Metaculus/metaculus/issues/1480

The title says it all really. Most my time was spent in evaluating different filters and aggregation methods. See this slack thread for some resulting graphs, conclusions, and discussion: https://metaculus.slack.com/archives/C52PD5LUS/p1733276259333379?thread_ts=1731959762.059169&cid=C52PD5LUS

I ended up spending more time on this than I should have. With extra time, I would add unit tests and spot check the calibration curve calculations as garunteeing this math works is important. However manual inspection of the math looks good. I probably should have switched to spam detection task once I realized that the resolution date filter did not fix things.

Additionally I wasn't able to get the current calibration graphs to exactly match the graphs from the old Metaculus site. There could be a number of confounding factors (e.g. slightly different but valid calculations, new questions, etc) so not sure what to do about that. This is another reason unit testing would be good.

### CODE REVIEW

File: users/services/spam_detection.py [hunk 1]
```diff
@@ -57,8 +57,8 @@ def check_profile_update_for_spam(
     duration = end_time - start_time
 
     if idenficated_as_spam:
-        logger.warning(
-            f"User {user.username} was soft deleted for spam bio: {bio_plus_website[:100]}... "
+        logger.info(
+            f"User: {user.username} ID: {user.id} was soft deleted for spam bio: {bio_plus_website[:100]}... "
```
Comment (lsabor) [inline users/services/spam_detection.py:61]: sneaking that change in there. :)
No problem

File: users/views.py
Comment (lsabor) [inline users/views.py:1]: Looks like you haven't changed any of the math. I guess it passed the smell test at least?
If/when we come back around to this general task, it's probably about time to go and scrounge through the old codebase data to see the differences (not looking foward to that)

**Review by lsabor — APPROVED** [review]
Great.
Happy to merge as-is (after pulling main and making sure tests pass).
Also doesn't need re-review if you make minor changes, it's solid.

File: users/views.py
Comment (CodexVeritas) [reply users/views.py:1]: Yep! It looked good. I experimented with an alternative, but decided your way was better.

**Pull request merged**